homeassistant/components/advantage_air/climate.py

```diff
@@ -248,10 +246,34 @@ async def async_set_hvac_mode(self, hvac_mode):
 
     async def async_set_temperature(self, **kwargs):
         """Set the Temperature."""
-        temp = kwargs.get(ATTR_TEMPERATURE)
-        await self.async_change(
-            {self.ac_key: {"zones": {self.zone_key: {"setTemp": temp}}}}
-        )
+        if ATTR_TEMPERATURE in kwargs:
+            await self.async_change(
+                {
+                    self.ac_key: {
+                        "zones": {self.zone_key: {"setTemp": kwargs[ATTR_TEMPERATURE]}}
+                    }
+                }
+            )
+            return
+        if ATTR_TARGET_TEMP_LOW in kwargs and ATTR_TARGET_TEMP_HIGH in kwargs:
+            await self.async_change(
+                {
+                    self.ac_key: {
+                        "zones": {
+                            self.zone_key: {
+                                "setTemp": int(
+                                    (
+                                        kwargs[ATTR_TARGET_TEMP_LOW]
+                                        + kwargs[ATTR_TARGET_TEMP_HIGH]
+                                    )
+                                    / 2
+                                )
+                            }
+                        }
+                    }
+                }
+            )
+            return
```

Comment:
@balloob is the an appropriate way to handle the possibility of having the set temp or the target temp ranges?